### ISSUE TITLE
AggregateActor "pipeTo self" bypasses termination logic of AggregateManager

### ISSUE THREAD

**Issue by rigolepe** [strongtyped/fun-cqrs#88]
Hi,

During stress-testing on fun.cqrs, we frequently see the following scenario:

```
< stress testing going on for a while, and then >
manager: create actor-1
manager: forward cmd-1 to actor-1 mailbox
actor-1 state recovering from DB
manager: create actor-2 => max children exceeded 
manager: cmd-kill to 20 children including actor-1
actor-1 state recovered from DB
actor-1 processing cmd-1 =>  cmd-success pipeTo self 
actor-1 processing cmd-kill 
actor-1 processing cmd-success !!! FAILS !!! => timeout on ask from REST controller 
```

The bottom-line is that the `pipeTo self` in following code in the AggregateActor is not being routed through the termination logic in `def forward(message: Any)` of the AggregateManager. This results in the piped command not being cached for processing in the new AggregateActor. 

```
        eventWithTimeout map {
          case (events, nextState) => Successful(events, nextState, origSender)
        } recover {
          case NonFatal(cause) => FailedCommand(cmd, cause, origSender)
        } pipeTo self
```

So, the solution would be to send the Successful message through the forwarding logic of the manager instead of using `pipeTo self`, so that these messages to actors that are being terminated are cached. 

Apart from that, it would be better to kill the oldest aggregate actors first. We see in our tests that the most recently created aggregate actor is killed when only 20 of the 40 actors are selected for recycling. 

Let me know if you want me to propose a solution in a merge request. 

Cheers,
   Peter

**Comment by rigolepe** [comment]
After an extensive search, the solution was super-simpel. See merge request https://github.com/strongtyped/fun-cqrs/pull/89

**Comment by rcavalcanti** [comment]
fixed by #89

**Comment by rcavalcanti** [comment]
Thanks @rigolepe
